src/trajectory.jl

```diff
@@ -134,67 +60,141 @@ end
 Slice variable is initialized.
 """
-Slice(rng::AbstractRNG, z0::PhasePoint) = Slice(z0, log(rand(rng)) - energy(z0), 1)
+SliceTS(rng::AbstractRNG, z0::PhasePoint) = SliceTS(z0, log(rand(rng)) - energy(z0), 1)
 
 """
 Multinomial sampler for the starting single leaf tree.
 (Log) weights for leaf nodes are their (unnormalised) Hamiltonian energies.
 
 Ref: https://github.com/stan-dev/stan/blob/develop/src/stan/mcmc/hmc/nuts/base_nuts.hpp#L226
 """
-Multinomial(rng::AbstractRNG, z0::PhasePoint) = Multinomial(z0, zero(energy(z0)))
+MultinomialTS(rng::AbstractRNG, z0::PhasePoint) = MultinomialTS(z0, zero(energy(z0)))
 
 """
 Create a slice sampler for a single leaf tree:
 - the slice variable is copied from the passed-in sampler `s` and
 - the number of acceptable candicates is computed by comparing the slice variable against the current energy.
 """
-Slice(s::Slice, H0::AbstractFloat, zcand::PhasePoint) = 
-    Slice(zcand, s.ℓu, (s.ℓu <= -energy(zcand)) ? 1 : 0)
+SliceTS(s::SliceTS, H0::AbstractFloat, zcand::PhasePoint) =
+    SliceTS(zcand, s.ℓu, (s.ℓu <= -energy(zcand)) ? 1 : 0)
 
 """
 Multinomial sampler for a trajectory consisting only a leaf node.
 - tree weight is the (unnormalised) energy of the leaf.
 """
-Multinomial(s::Multinomial, H0::AbstractFloat, zcand::PhasePoint) = 
-    Multinomial(zcand, H0 - energy(zcand))
+MultinomialTS(s::MultinomialTS, H0::AbstractFloat, zcand::PhasePoint) =
+    MultinomialTS(zcand, H0 - energy(zcand))
 
-function combine(rng::AbstractRNG, s1::Slice, s2::Slice)
+function combine(rng::AbstractRNG, s1::SliceTS, s2::SliceTS)
     @assert s1.ℓu == s2.ℓu "Cannot combine two slice sampler with different slice variable"
     n = s1.n + s2.n
     zcand = rand(rng) < s1.n / n ? s1.zcand : s2.zcand
-    Slice(zcand, s1.ℓu, n)
+    SliceTS(zcand, s1.ℓu, n)
 end
 
-function combine(zcand::PhasePoint, s1::Slice, s2::Slice)
+function combine(zcand::PhasePoint, s1::SliceTS, s2::SliceTS)
     @assert s1.ℓu == s2.ℓu "Cannot combine two slice sampler with different slice variable"
     n = s1.n + s2.n
-    Slice(zcand, s1.ℓu, n)
+    SliceTS(zcand, s1.ℓu, n)
 end
 
-function combine(rng::AbstractRNG, s1::Multinomial, s2::Multinomial)
+function combine(rng::AbstractRNG, s1::MultinomialTS, s2::MultinomialTS)
     ℓw = logaddexp(s1.ℓw, s2.ℓw)
     zcand = rand(rng) < exp(s1.ℓw - ℓw) ? s1.zcand : s2.zcand
-    Multinomial(zcand, ℓw)
+    MultinomialTS(zcand, ℓw)
 end
 
-function combine(zcand::PhasePoint, s1::Multinomial, s2::Multinomial)
+function combine(zcand::PhasePoint, s1::MultinomialTS, s2::MultinomialTS)
     ℓw = logaddexp(s1.ℓw, s2.ℓw)
-    Multinomial(zcand, ℓw)
+    MultinomialTS(zcand, ℓw)
 end
 
 mh_accept(
     rng::AbstractRNG,
-    s::Slice,
-    s′::Slice
+    s::SliceTS,
+    s′::SliceTS
 ) = rand(rng) < min(1, s′.n / s.n)
 
 mh_accept(
     rng::AbstractRNG,
-    s::Multinomial,
-    s′::Multinomial
+    s::MultinomialTS,
+    s′::MultinomialTS
 ) = rand(rng) < min(1, exp(s′.ℓw - s.ℓw))
 
+"""
```

Comment:
Is this new block just a copy and paste?

Reply:
Yes. I just moved the code to the very begining (before static trajectires). The reason is we later would like to reuse these struct for static trajecotries as well.